test/e2e/webdriver/driver.js

```diff
@@ -884,98 +895,27 @@ class Driver {
-    title,
-    initialWindowHandles = null,
-    delayStep = 1000,
-    timeout = this.timeout,
-    { retries = 8, retryDelay = 2500 } = {},
-  ) {
-    let windowHandles =
-      initialWindowHandles || (await this.driver.getAllWindowHandles());
-    let timeElapsed = 0;
-
-    while (timeElapsed <= timeout) {
-      for (const handle of windowHandles) {
-        const handleTitle = await retry(
-          {
-            retries,
-            delay: retryDelay,
-          },
-          async () => {
-            await this.driver.switchTo().window(handle);
-            return await this.driver.getTitle();
-          },
-        );
-
-        if (handleTitle === title) {
-          return handle;
-        }
-      }
-      await this.delay(delayStep);
-      timeElapsed += delayStep;
-      // refresh the window handles
-      windowHandles = await this.driver.getAllWindowHandles();
-    }
-
-    throw new Error(`No window with title: ${title}`);
+  async switchToWindowWithTitle(title) {
+    return await this.windowHandles.switchToWindowWithProperty('title', title);
   }
 
   /**
    * Switches the context of the browser session to the window tab with the given URL.
    *
-   * @param {string} url - Window URL to switch
-   * @param {string} [initialWindowHandles] - optional array of window handles to search through.
-   * If not provided, the function fetches all current window handles.
-   * @param {int} delayStep - optional defaults to 1000 milliseconds
-   * @param {int} timeout - optional set to the defaults to 1000 milliseconds in the file
-   * @param {int} retries,retryDelay - optional for retrying the URL fetch operation, defaults to starting at 8 ms to 2500 ms
+   * @param {string} url - Window URL to find
    * @returns {Promise<void>}  promise that resolves once the switch is complete
-   * @throws {Error} throws an error if no window with the specified url is found
+   * @throws {Error} throws an error if no window with the specified URL is found
    */
-  async switchToWindowWithUrl(
-    url,
-    initialWindowHandles,
-    delayStep = 1000,
-    timeout = this.timeout,
-    { retries = 8, retryDelay = 2500 } = {},
-  ) {
-    let windowHandles =
-      initialWindowHandles || (await this.driver.getAllWindowHandles());
-    let timeElapsed = 0;
-
-    while (timeElapsed <= timeout) {
-      for (const handle of windowHandles) {
-        const handleUrl = await retry(
-          {
-            retries,
-            delay: retryDelay,
-          },
-          async () => {
-            await this.driver.switchTo().window(handle);
-            return await this.driver.getCurrentUrl();
-          },
-        );
-
-        if (handleUrl === `${url}/`) {
-          return handle;
-        }
-      }
-      await this.delay(delayStep);
-      timeElapsed += delayStep;
-      // refresh the window handles
-      windowHandles = await this.driver.getAllWindowHandles();
+  async switchToWindowWithUrl(url) {
+    let _url = url;
```

Comment:
I loathe the eslint rule that necessitates this nonsense.

Reply:
Oh yes I definitely agree with you.  Do you think we have Consensys/consensus to turn it off?